src/AdminConsole/Services/SetupService.cs

```diff
@@ -1,18 +1,17 @@
-using Microsoft.Extensions.Options;
 using Passwordless.Common.Configuration;
 
 namespace Passwordless.AdminConsole.Services;
 
 public class SetupService : ISetupService
 {
-    private readonly PasswordlessOptions _options;
+    private readonly PasswordlessOptions? _options;
     private readonly bool _isSelfHosted;
 
     public SetupService(
-        IOptions<PasswordlessOptions> options,
         IConfiguration configuration)
     {
-        _options = options.Value;
+        // Get the Passwordless options manually to skip the built-in validations in the Passwordless SDK
+        _options = configuration.GetSection("Passwordless").Get<PasswordlessOptions>();
```

Comment:
Use IOptionsSnapshot.

https://learn.microsoft.com/en-us/aspnet/core/fundamentals/configuration/options?view=aspnetcore-8.0#use-ioptionssnapshot-to-read-updated-data

Reply:
It's not possible to use `IOptionsSnapshot` for the reason explained in the comment.

Reply:
@Tyrrrz There is nothing wrong with the order of the configurations. I can help you with it next week.

Line 79 in:
```src/AdminConsole/Db/DatabaseBootstrap.cs```

Take a look at how it's configured. The lambda expression doesn't support `reloadOnChange`.

Change it to `.AddPasswordless("Passwordless")` and it would work just fine both in Docker and using the SelfHosting run configuration.

However doing so would break mapping the `InternalApiUrl`. So we'd probably have to fix this.

See also:
- https://github.com/bitwarden/passwordless-dotnet/pull/124
- https://github.com/bitwarden/passwordless-dotnet/pull/125

`InternalApiUrl` under `PasswordlessManagement` was meant to prevent network traffic to unnecessarily leave the container.

Reply:
@jonashendrickx I think you're a bit confused in regard to what is the actual problem here. Reloading works completely fine. `IOptionsSnapshot<>` would work fine too, if not for how `AddPasswordless()` works.

The core issue is here:

https://github.com/bitwarden/passwordless-dotnet/blob/aa04a1d7a981b9b998e1c45e20c32937c5aab983/src/Passwordless/ServiceCollectionExtensions.cs#L22-L25

As you can see, the options are created with a validator that checks that the `ApiSecret` is not null or empty. In other words, when we call `AddPasswordless`, the secret must already have been set.

However, during the initialization sequence, it is not set yet. The whole purpose of the initialization sequence is to generate the `ApiSecret` and add it to the configuration. So in order to check whether it has been added yet or not, we can't rely on the `PasswordlessOptions` provided by the Passwordless SDK since they would fail to initialize immediately, seeing as `ApiSecret` is null.

Reply:
Seems like whatever solution we would go with it would be personal preference.